### PR TITLE
Adds Operations for Creating Points and Sizes and Accessing Fields on Mats

### PR DESCRIPTION
**Pull request by JLLeitschuh**

### CODE REVIEW

File: src/main/java/edu/wpi/grip/core/operations/opencv/MatFieldAccessor.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import edu.wpi.grip.core.OutputSocket;
+import edu.wpi.grip.core.SocketHint;
+import org.bytedeco.javacpp.opencv_core.Mat;
+import org.bytedeco.javacpp.opencv_core.Size;
+
+public class MatFieldAccessor implements CVOperation{
+    private static final Mat defaultsMat = new Mat();
+    private final SocketHint matHint = new SocketHint("input", Mat.class, Mat::new, SocketHint.View.NONE);
+    private final SocketHint sizeHint = new SocketHint("size", Size.class, Size::new);
+    private final SocketHint isNullHint = new SocketHint("is null", Boolean.class, defaultsMat.isNull());
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/operations/opencv/MatFieldAccessor.java]: Mats shouldn't ever be `null`.  If this returns true, I'm pretty sure something in the application would crash (or more likely OpenCV would segfault), so having this as an output is not useful.

File: src/main/java/edu/wpi/grip/core/operations/opencv/MatFieldAccessor.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    private final SocketHint emptyHint = new SocketHint("empty", Boolean.class, defaultsMat.empty());
+    private final SocketHint channelsHint = new SocketHint("channels", Number.class, defaultsMat.channels());
+    private final SocketHint colsHint = new SocketHint("cols", Number.class, defaultsMat.cols());
+    private final SocketHint rowsHint = new SocketHint("rows", Number.class, defaultsMat.rows());
+    private final SocketHint highValueHint = new SocketHint("high value", Number.class, defaultsMat.highValue());
+    private final SocketHint depthHint = new SocketHint("depth", Number.class, defaultsMat.depth());
+    private final SocketHint typeHint = new SocketHint("type", Number.class, defaultsMat.type());
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/operations/opencv/MatFieldAccessor.java]: This is an enum, not a Number

File: src/main/java/edu/wpi/grip/core/operations/opencv/MatFieldAccessor.java (hunk 1 above)
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/core/operations/opencv/MatFieldAccessor.java]: Makes sense.

File: src/main/java/edu/wpi/grip/core/operations/opencv/PointGeneratorOperation.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    private final SocketHint<Point> outputHint = new SocketHint<Point>("point", Point.class, Point::new, SocketHint.View.NONE);
+
+    @Override
+    public String getName() {
+        return "Point";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Creates a Point";
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/operations/opencv/PointGeneratorOperation.java]: I would name this "New Point" and word the description as "Create a point".  Same for Size.

File: src/main/java/edu/wpi/grip/core/operations/opencv/PointGeneratorOperation.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+package edu.wpi.grip.core.operations.opencv;
+
+
+import com.google.common.eventbus.EventBus;
+import edu.wpi.grip.core.InputSocket;
+import edu.wpi.grip.core.OutputSocket;
+import edu.wpi.grip.core.SocketHint;
+import org.bytedeco.javacpp.opencv_core.Point;
+
+public class PointGeneratorOperation implements CVOperation {
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/operations/opencv/PointGeneratorOperation.java]: I think `NewPointOperation` would be a better name for this class (and similarly for the Size one)

File: src/main/java/edu/wpi/grip/core/operations/opencv/SizeGeneratorOperation.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+import com.google.common.eventbus.EventBus;
+import edu.wpi.grip.core.InputSocket;
+import edu.wpi.grip.core.OutputSocket;
+import edu.wpi.grip.core.SocketHint;
+import org.bytedeco.javacpp.opencv_core.Size;
+
+public class SizeGeneratorOperation implements CVOperation {
+
+    private final SocketHint<Number> widthHint = new SocketHint<Number>("width", Number.class, -1, SocketHint.View.SPINNER, new Integer[]{Integer.MIN_VALUE, Integer.MAX_VALUE}, false);
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/operations/opencv/SizeGeneratorOperation.java]: Should the minimum value be 0?  I don't think negative sizes make sense, and OpenCV will probably spit back a bunch of errors if we try to to that.

Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/core/operations/opencv/SizeGeneratorOperation.java]: Perhaps the minimum should be -1. The default for a lot of opencv functions is `Point(-1,-1)`

File: src/main/java/edu/wpi/grip/core/operations/opencv/MatFieldAccessor.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    private final SocketHint emptyHint = new SocketHint("empty", Boolean.class, defaultsMat.empty());
+    private final SocketHint channelsHint = new SocketHint("channels", Number.class, defaultsMat.channels());
+    private final SocketHint colsHint = new SocketHint("cols", Number.class, defaultsMat.cols(), SocketHint.View.NONE, null, true);
+    private final SocketHint rowsHint = new SocketHint("rows", Number.class, defaultsMat.rows(), SocketHint.View.NONE, null, true);
+    private final SocketHint highValueHint = new SocketHint("high value", Number.class, defaultsMat.highValue());
+
+
+    @Override
+    public String getName() {
+        return "Unwrap Mat";
```
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/core/operations/opencv/MatFieldAccessor.java]: Get Mat Info

**Pull request merged**